examples/bloom_filter/host_bulk_example.cu

```diff
@@ -25,44 +25,68 @@
 
 int main(void)
 {
-  // Generate 10'000 keys and insert the first 5'000 into the filter.
-  int constexpr num_keys = 10'000;
-  int constexpr num_tp   = num_keys * 0.5;
-  int constexpr num_tn   = num_keys - num_tp;
+  using key_type    = int;
+  using extent_type = size_t;
 
-  // Spawn a filter with 200 sub-filters.
-  cuco::bloom_filter<int> filter{200};
+  // Generate 10'000 keys and insert the first 5'000 into the filter containing of 200 sub-filters.
+  int constexpr num_keys    = 10'000;
+  int constexpr num_tp      = num_keys * 0.5;
+  int constexpr num_tn      = num_keys - num_tp;
+  int constexpr sub_filters = 200;
 
-  thrust::device_vector<int> keys(num_keys);
-  thrust::sequence(keys.begin(), keys.end(), 1);
+  // Lambda function to bulk insert to the provided bloom filter and evaluate FPR.
+  auto bulk_insert_and_evaluate_bloom_filter = [&](auto& filter) {
```

Comment:
Templated lambdas are only available in C++20. Can we add the `-std=c++2a` flag to this example or should I just use a normal templated function/functor for this.

Reply:
Let's use templated functor instead